x-pack/elastic-agent/magefile.go

```diff
@@ -591,6 +591,9 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 				cmd.Stdout = os.Stdout
 				cmd.Stderr = os.Stderr
 				cmd.Env = append(os.Environ(), fmt.Sprintf("PWD=%s", pwd), "AGENT_PACKAGING=on")
+				if len(devtools.SelectedPackageTypes) == 1 && devtools.SelectedPackageTypes[0] == devtools.Docker {
+					cmd.Env = append(cmd.Env, "PACKAGES=targz")
```

Comment:
Should we do this in any case if docker is one of the selected packages? also if there are more than one selected package.

Reply:
ATM it's not necessary. But for the sake of being future proof, I have rewritten the code.